packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/image/ImageLoaderModule.kt

```diff
@@ -227,16 +230,17 @@ internal class ImageLoaderModule : NativeImageLoaderAndroidSpec, LifecycleEventL
     @Suppress("DEPRECATION", "StaticFieldLeak")
     object : GuardedAsyncTask<Void, Void>(getReactApplicationContext()) {
           override fun doInBackgroundGuarded(vararg params: Void) {
-            val result: WritableMap = Arguments.createMap()
-            val imagePipeline: ImagePipeline = this@ImageLoaderModule.imagePipeline
-            for (i in 0 until uris.size()) {
-              val uriString = uris.getString(i)
-              if (!uriString.isNullOrEmpty()) {
-                val uri = Uri.parse(uriString)
-                if (imagePipeline.isInBitmapMemoryCache(uri)) {
-                  result.putString(uriString, "memory")
-                } else if (imagePipeline.isInDiskCacheSync(uri)) {
-                  result.putString(uriString, "disk")
+            val result = buildReadableMap {
+              val imagePipeline: ImagePipeline = this@ImageLoaderModule.imagePipeline
+              repeat(uris.size()) {
+                val uriString = uris.getString(it)
+                if (!uriString.isNullOrEmpty()) {
+                  val uri = uriString.toUri()
```

Comment:
And also those

Reply:
Apply on https://github.com/facebook/react-native/pull/51145/commits/ec3af404a121ccb79b2d8181d3647700b0e398e8